app/user_management/roles/routers.py

```diff
@@ -1,105 +1,54 @@
-from fastapi import APIRouter, Depends, HTTPException
+from fastapi import APIRouter, Depends
 from fastapi.responses import JSONResponse
 
 from app.roles_handler.roles_manager import verify_current_user
 from app.roles_handler.roles_schema import ApplicationRoles, OrganisationRoles
+from app.user_management.roles.roles_service_manager import get_roles_service_manager, RolesServiceManager
 from app.user_management.roles.schemas import CreateRoleFields, UpdateRoleFields
-from app.utils.request_exceptions import (
-    BaseApiException,
-    ServiceUnavailableException,
-    BadRequestException
-)
-from app.utils.request_handler import RequestHandler, get_request_handler
+
 
 router = APIRouter(prefix="/api/v1/roles")
 
 
+
 @router.post("/")
 async def create_role(
+        query_parameters: CreateRoleFields = Depends(),
         user_role: dict = Depends(verify_current_user(
-            [
-                ApplicationRoles.APPLICATION_ADMIN.value,
-            ]
+            [ApplicationRoles.APPLICATION_ADMIN.value]
         )),
-        query_parameters: CreateRoleFields = Depends(),
-        request_handler: RequestHandler = Depends(get_request_handler)
+        roles_service_manager: RolesServiceManager = Depends(get_roles_service_manager)
 ):
-    try:
-        users_info = await request_handler.make_request(
-            method="POST",
-            endpoint='http://host.docker.internal:8001/api/v1/roles/',
-            content=query_parameters.model_dump_json(exclude_none=True)
-        )
-        return JSONResponse(content=users_info)
-    except BadRequestException as e:
-        raise HTTPException(
-            status_code=400,
-            detail=str(e)
-        )
-    except (ServiceUnavailableException, BaseApiException):
-        raise HTTPException(
-            status_code=500,
-            detail="Service unavailable"
-        )
+    users_info = await roles_service_manager.create_role(query_parameters.model_dump(exclude_none=True))
+    return JSONResponse(content=users_info)
 
 
 @router.delete("/{role_id}")
 async def delete_role(
         role_id: str,
         user_role: dict = Depends(verify_current_user(
-            [
-                ApplicationRoles.APPLICATION_ADMIN.value,
-            ]
+            [ApplicationRoles.APPLICATION_ADMIN.value]
         )),
-        request_handler: RequestHandler = Depends(get_request_handler)
+        roles_service_manager: RolesServiceManager = Depends(get_roles_service_manager)
```

Comment:
hint:
you can use this annotation `role_service_manager: GetRoleServiceManager` if you implement a proper annotation for that, e.g

`GetEpochsSubgraph = Annotated[
    EpochsSubgraph,
    Depends(get_epochs_subgraph),
]`